e2e/cypress/integration/archived_channel/archive_channel_search_spec.js

```diff
@@ -0,0 +1,163 @@
+    let testTeam;
+    let testChannel;
+    let testUser;
+    let adminUser;
+    const testArchivedMessage = `this is an archived post ${getRandomId()}`;
+
+    before(() => {
+        adminUser = getAdminAccount();
+
+        cy.apiUpdateConfig({
+            TeamSettings: {
+                ExperimentalViewArchivedChannels: true,
+            },
+        });
+
+        // # Login as test user and visit town-square
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            testTeam = team;
+            testChannel = channel;
+            testUser = user;
+
+            cy.apiCreateUser({prefix: 'second'}).then(({user: second}) => {
+                cy.apiAddUserToTeam(testTeam.id, second.id);
+            });
+            cy.visit(`/${team.name}/channels/${testChannel.name}`);
+            cy.postMessageAs({sender: testUser, message: testArchivedMessage, channelId: testChannel.id});
+        });
+    });
+
+    it('MM-T1707 Unarchived channels can be searched the same as before they where archived', () => {
+        cy.apiLogin(adminUser);
+
+        // # Post some text in the channel such as "Pineapple"
+        const messageText = `pineaple ${getRandomId()}`;
+
+        // # Archive the channel
+        createArchivedChannel({prefix: 'pineaple-'}, [messageText]).then(() => {
+            // # Unarchive the channel:
+            cy.uiUnarchiveChannel().then(() => {
+                // # remove search on archived channels
+                cy.apiUpdateConfig({
+                    TeamSettings: {
+                        ExperimentalViewArchivedChannels: false,
+                    },
+                });
+
+                cy.visit(`/${testTeam.name}/channels/off-topic`);
+                cy.contains('#channelHeaderTitle', 'Off-Topic');
+
+                // # Search for the post from step 1')
+                cy.get('#searchBox').focus().clear();
+                cy.get('#searchBox').should('be.visible').type(`${messageText}{enter}`);
+
+                // * Post is returned by search, since it's not archived anymore
+                cy.get('#searchContainer').should('be.visible');
+                cy.get('.search-item-snippet').first().contains(messageText);
+            });
+        });
+    });
+    it('MM-T1708 An archived channel can\'t be searched when "Allow users to view archived channels" is set to False in "Site Configuration > Users and Teams" in the System Console', () => {
+    // # First, as system admin, ensure that System Console > Users and Teams > Allow users to view archived channels is set to `false`.
+        cy.apiUpdateConfig({
+            TeamSettings: {
+                ExperimentalViewArchivedChannels: false,
+            },
+        });
+
+        cy.apiLogin(testUser);
+
+        // # Open a channel other than Town Square
+        cy.visit(`/${testTeam.name}/channels/off-topic`);
+        cy.contains('#channelHeaderTitle', 'Off-Topic');
+
+        // next steps were performed previously
```

Comment:
nit: do we need this extra comment?

Reply:
that was in case someone was following the test script, so they know the steps were performed in the `before` section. Should I remove it?

Reply:
I am not sure either, just wen't about the other files' convention 🤷‍♀️  I guess we can keep it or wait for Saturn's feedback

Reply:
Looks like [posting "testArchivedMessage"](https://github.com/mattermost/mattermost-webapp/pull/7205/files#diff-ca2598e8a2a52dfd97792d31a2f326c32399836f93c14b92d59ac5b40d27fc0eR42) is actually intended in this `it` block only. If yes, then I suggest to move the line here and collocate with the comment so it's easier to read/follow.